python/ray/actor.py

```diff
@@ -691,17 +700,22 @@ def _serialization_helper(self, ray_forking):
             "actor_method_names": self._ray_actor_method_names,
             "method_signatures": self._ray_method_signatures,
             "method_num_return_vals": self._ray_method_num_return_vals,
-            # Actors in local mode don't have dummy objects.
             "actor_creation_dummy_object_id": self.
-            _ray_actor_creation_dummy_object_id.id()
-            if self._ray_actor_creation_dummy_object_id is not None else None,
+            _ray_actor_creation_dummy_object_id.id(),
             "actor_method_cpus": self._ray_actor_method_cpus,
             "actor_driver_id": self._ray_actor_driver_id.id(),
             "ray_forking": ray_forking
         }
 
         if ray_forking:
             self._ray_actor_forks += 1
+        # Notify the backend to expect this new actor handle. The backend will
+        # not release the cursor for any new handles until the first task for
+        # each of the new handles is submitted.
+        # NOTE(swang): If the new actor handle fails to be used (e.g., due
+        # to a failure to register a named actor), then this may cause a
```

Comment:
This could happen even without any failure to register a named actor, right? E.g.,

```python
a = Actor.remote()

@ray.remote
def f(a):
    return

f.remote(a)
```

What precisely gets leaked? The dummy object ID that the forked ID depends on?

Reply:
Hmm yeah, I guess the more general problem is GC for actor handles.